clients/src/main/java/org/apache/kafka/common/requests/MetadataResponse.java

```diff
@@ -72,13 +76,24 @@
     private static final String ISR_KEY_NAME = "isr";
 
     private final Collection<Node> brokers;
+    private final Node controller;
     private final List<TopicMetadata> topicMetadata;
 
+    /**
+     * Constructor for the latest version
+     */
+    public MetadataResponse(List<Node> brokers, int controllerId, List<TopicMetadata> topicMetadata) {
```

Comment:
I actually like the deprecated constructors because we get explicit compile warnings when we bump versions and don't accidentally forget to update some of the places where they are called. 

If you are not convinced that having warnings to help with bumps is useful, I won't make you change it ;)